questionnaire_linking.py

```diff
@@ -0,0 +1,117 @@
+import csv
+import argparse
+import requests
+import qid_checksum_validator
+from config import Config
+from datetime import datetime
+import uuid
+import json
+from utilities.rabbit_context import RabbitContext
+
+
+def validate_qid_link_file(qid_link_file_path):
+    try:
+        with open(qid_link_file_path, encoding="utf-8") as qid_link_file:
+            return qid_file(qid_link_file)
+    except UnicodeDecodeError as err:
+        print(f'Invalid file encoding, requires utf-8, error: {err}')
+        exit(-1)
+
+
+def qid_file(qid_link_file):
+    qid_link_file_reader = csv.DictReader(qid_link_file, delimiter=',', fieldnames=['case_ref', 'qid'])
+    validate_and_submit_questionnaire_links(qid_link_file_reader)
+
+
+def parse_arguments():
+    parser = argparse.ArgumentParser(description='Load a QID Link file into response management.')
+    parser.add_argument('qid_link_file_path', help='path to the QID Link file', type=str)
+    return parser.parse_args()
+
+
+args = parse_arguments()
+error_count = 0
+
+
+def validate_and_submit_questionnaire_links(qid_link_file_reader):
```

Comment:
should this show an error for an empty file?

Reply:
Sorted